Expound on the view: fields

cvalarida opened the pull request:
Gives another example of what the `view:` fields can be used for.

cvalarida committed:
Expound on the view: fields

Gives another example of what the `view:` fields can be used for.

cvalarida commented on docs/schemaform/form-config.md:
```diff
@@ -66,9 +66,23 @@ Forms are created by creating a page that uses FormApp from the schemaform folde
           // These can be used to remove fields from what is sent to the server, like if we have
-          // a question that is only used to reveal other questions
+          // a question that is only used to reveal other questions or group related questions
+          // together to be conditionally revealed that aren't in an object in the schema
```
Perhaps this could be explained better? I'm thinking for situations like the 5490's `previousBenefits` page where some options (`chapter35`, `transferOfEntitlement`, etc.`) were revealed under `view:claimedSponsorService`, but siblings of the other options (`chapter31`, `disability`, etc.) in the schema. In that case, I was able to `expandUnder` for each of them, but if `hideIf` is required, these could be grouped in a `view:someName` and the `hideIf` applied to `view:someName` only without affecting the end result sent to the api.

Does this explanation make the above use case clear?

jbalboni replied:
We could add use cases to the cookbook section. Like, "I want to conditionally hide a group of fields" and "I want to group fields together to validate them together" (used for email on the contact info page).

cvalarida commented on docs/schemaform/form-config.md:
```diff
@@ -66,9 +66,23 @@ Forms are created by creating a page that uses FormApp from the schemaform folde
+            type: 'object',
+            properties: {
+              // view:artificialGroup will be flattened; subField1 and subField2 will be
+              // siblings of field1 when sent to the api
```
If the description above doesn't make it clear, I hope this does the trick.

cvalarida committed:
Add cookbook recipe

I'm not sure if it's much clearer this way, but it does give a more detailed example.

jbalboni commented on docs/schemaform/form-config.md:
````diff
@@ -352,3 +366,61 @@ You can use 'ui:description' to show text or a custom component before the field
+  },
+  other: { ... }
+}
+```
````
Should add a line here about the backend not seeing the `view:` object.

"When this form is sent to the backend, the fields in the `view:sponsorServiceOptions` object will be moved up one level, i.e. they'll be sent alongside `dic` and `chapter31`. The backend will never see objects with names that start with `view:`, but it will get all the fields inside of them."

Or something similar. It's maybe redundant, but it can't hurt.

cvalarida replied:
I like that.

Done!

jbalboni reviewed (approved):
Made one suggestion, but LGTM.

raquelromano-gov commented on docs/schemaform/form-config.md:
````diff
@@ -352,3 +366,61 @@ You can use 'ui:description' to show text or a custom component before the field
   }
 }
 ```
+### I want to conditionally hide a group of fields
````
Super helpful to have the below example in addition to the explanation above!

cvalarida committed:
Add conclusive statement about the back end
